kustomize/commands/internal/kustfile/kustomizationfile.go

```diff
@@ -235,7 +241,23 @@ func (mf *kustomizationFile) marshal(kustomization *types.Kustomization) ([]byte
 		if err != nil {
 			return content, err
 		}
-		output = append(output, content...)
+
+		trimOriginalLine := bytes.TrimSpace(comment.originalLine)
+		trimContent := bytes.TrimSpace(content)
+
+		deprecatedFixes := []string{"vars:", "patchesJson6902:", "patchesStrategicMerge:", "commonLabels:"}
+
+		if !slices.Contains(deprecatedFixes, string(trimOriginalLine)) {
+			if comment.field == "ConfigMapGenerator" || comment.field == "SecretGenerator" {
+				continue
```

Comment:
I'm a bit curious about this part. Is it the case that `ConfigMapGenerator`s and `SecretGenerator`s will not deal well with comments, or we just don't want to retain comments on these two elements?